plugins/com.google.cloud.tools.eclipse.appengine.login.test/src/com/google/cloud/tools/eclipse/appengine/login/ui/LoginServiceUiTest.java

```diff
@@ -103,8 +103,13 @@ public void run() {
         ) {
           String input = new String();
           while (true) {
-            input += (char) reader.read();
-            if (input.endsWith("?error=cancelled-by-user")) {
+            int ch = reader.read();
+            if (ch == -1) {
+              break;
```

Comment:
You can write this a little more simply as a for loop:

for (int ch = reader.read(); ch != -1; ch = reader.read())